app/Jobs/WikiEntityImportJob.php

```diff
@@ -0,0 +1,240 @@
+use Illuminate\Bus\Queueable;
+use Illuminate\Contracts\Queue\ShouldQueue;
+use Illuminate\Foundation\Bus\Dispatchable;
+use Illuminate\Queue\InteractsWithQueue;
+use Illuminate\Queue\SerializesModels;
+use Illuminate\Support\Facades\Http;
+use Illuminate\Support\Facades\Log;
+use Illuminate\Support\Facades\Config;
+use App\Wiki;
+use App\WikiEntityImport;
+use Carbon\Carbon;
+use Maclof\Kubernetes\Client;
+use Maclof\Kubernetes\Models\Job as KubernetesJob;
+
+class WikiEntityImportJob implements ShouldQueue
+{
+    use Dispatchable, InteractsWithQueue, Queueable, SerializesModels;
+
+    /**
+     * Create a new job instance.
+     */
+    public function __construct(
+        public int $wikiId,
+        public string $sourceWikiUrl,
+        public array $entityIds,
+        public int $importId,
+    )
+    {}
+
+    private string $targetWikiUrl;
+
+    /**
+     * Execute the job.
+     */
+    public function handle(Client $kubernetesClient): void
+    {
+        $import = null;
+        try {
+            $wiki = Wiki::findOrFail($this->wikiId);
+            $import = WikiEntityImport::findOrFail($this->importId);
+            $creds = $this->acquireCredentials($wiki->domain);
+
+            $this->targetWikiUrl = str_contains($wiki->domain, "localhost")
+                ? "http://".$wiki->domain
+                : "https://".$wiki->domain;
+
+            $kubernetesJob = new TransferBotKubernetesJob(
+                kubernetesClient: $kubernetesClient,
+                wiki: $wiki,
+                creds: $creds,
+                entityIds: $this->entityIds,
+                sourceWikiUrl: $this->sourceWikiUrl,
+                targetWikiUrl: $this->targetWikiUrl,
+                importId: $this->importId,
+            );
+            $jobName = $kubernetesJob->spawn();
+            Log::info(
+                'transferbot job for wiki "'.$wiki->domain.'" was created with name "'.$jobName.'".'
+            );
+        } catch (\Exception $ex) {
+            Log::error('Entity import job failed with error: '.$ex->getMessage());
+            $import?->update([
+                'status' => WikiEntityImportStatus::Failed,
+                'finished_at' => Carbon::now(),
+            ]);
+            $this->fail(
+                new \Exception('Error spawning transferbot for wiki '.$this->wikiId.': '.$ex->getMessage()),
+            );
+        }
+    }
+
+    private static function acquireCredentials(string $wikiDomain): OAuthCredentials
+    {
+        $response = Http::withHeaders(['host' => $wikiDomain])->asForm()->post(
+            getenv('PLATFORM_MW_BACKEND_HOST').'/w/api.php?action=wbstackPlatformOauthGet&format=json',
+            [
+                'consumerName' => 'WikiEntityImportJob',
+                'ownerOnly' => '1',
+                'consumerVersion' => '1',
+                'grants' => 'basic|highvolume|import|editpage|editprotected|createeditmovepage|uploadfile|uploadeditmovefile|rollback|delete|mergehistory',
```

Comment:
I'm pretty sure these could be reduced even further, but I am totally lost when debugging and was happy I found a working combo that is not "every permission ever".

Reply:
I'm surprised most of these are needed. I'm trying to think if there is a quick way for us to test these